fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayBanner.kt

```diff
@@ -325,47 +328,100 @@ private fun generateSingleSelectBannerMenuItems(
     }
 }
 
+private const val MAX_VISIBLE_TABS = 99
+private const val SO_MANY_TABS_OPEN = "∞"
+private val INFINITE_CHAR_TABS_PADDING_BOTTOM = 3.dp
+private val NUMBER_TABS_PADDING_BOTTOM = 1.dp
+
+private const val ONE_DIGIT_SIZE_RATIO = 0.5f
+private const val TWO_DIGITS_SIZE_RATIO = 0.4f
+
 @Composable
-private fun NormalTabsTabIcon(normalTabCount: Int) {
+@Suppress("MagicNumber")
+private fun NormalTabsTabIcon(normalTabCount: Int, selected: Boolean) {
     val normalTabCountText: String
     val normalTabCountTextModifier: Modifier
-    if (normalTabCount > TabCounter.MAX_VISIBLE_TABS) {
-        normalTabCountText = TabCounter.SO_MANY_TABS_OPEN
-        normalTabCountTextModifier = Modifier.padding(bottom = 1.dp)
-    } else {
-        normalTabCountText = normalTabCount.toString()
-        normalTabCountTextModifier = Modifier
+    val tabCountTextRatio: Float
+
+    when (normalTabCount) {
+        in 0..9 -> {
+            normalTabCountText = normalTabCount.toString()
+            tabCountTextRatio = ONE_DIGIT_SIZE_RATIO
+            normalTabCountTextModifier =
+                Modifier.padding(bottom = NUMBER_TABS_PADDING_BOTTOM)
+        }
+        in 10..MAX_VISIBLE_TABS -> {
+            normalTabCountText = normalTabCount.toString()
+            tabCountTextRatio = TWO_DIGITS_SIZE_RATIO
+            normalTabCountTextModifier =
+                Modifier.padding(
+                    bottom = NUMBER_TABS_PADDING_BOTTOM,
+                    end = NUMBER_TABS_PADDING_BOTTOM,
```

Comment:
it looks like the original `TabCounter` is not adding end padding. is there a reason it's needed here? I'm seeing a visual bug on my device with 99
<img width="397" alt="image" src="https://github.com/mozilla-mobile/firefox-android/assets/87384386/b1eb7457-c806-4f9e-9b14-dcc41ecaac8f">